### ISSUE TITLE
Scale filter makes the image as red with the "color blended layers" option

### ISSUE THREAD

**Issue by Vinzius** [Alamofire/AlamofireImage#65]
Hi !

First, thank you for this great library, I really appreciate its design.

I noticed that when I'm using a scaling method of AlamofireImage (e.g. `af_imageAspectScaledToFillSize`), I see its container (basic UIImageView) as red in the "color blended layers". But if I'm using my "own", then the color is green.

I'm putting below the code I used to create the image/imageView (nothing special):
```Swift
// instanciate image view
let imageView = UIImageView()
imageView.frame = CGRectMake(10.0, 10.0, 50.0, 50.0)
self.view.addSubview(imageView)

// add image
if let image = UIImage(named: "image-test")?.af_imageAspectScaledToFillSize(CGSizeMake(50.0, 50.0)) {
    image.af_inflate()
    imageView.image = image
}
```

I'm not sure if it's a big performance issue or not.
The resizing code I'm using is from this [blog](http://vocaro.com/trevor/blog/2009/10/12/resize-a-uiimage-the-right-way/), which I converted to Swift.

1/ Do you want an Xcode project to reproduce the issue?
2/ Do you think it would be worth updating the scaling method? I could manage a pull-request if you want.

Notes: Xcode 7.2, iOS 9.2

Cheers,
Vincent

**Comment by Vinzius** [comment]
It seems the problem is more simple than I anticipated.

The uiimageview is red because the scaled image isn't fully opaque.

The culprit is here:
```UIGraphicsBeginImageContextWithOptions(size, false, 0.0)```

Should we find a way to specify a fully opaque image?

**Comment by cnoon** [comment]
Thank you for reporting this @Vinzius!

I spent a bunch of time trying to figure out how to elegantly solve this problem. After several different approaches, I'm fairly happy with what I came up with. I added some alpha component inspection properties in 86602b26. Then I used those new properties to be smarter about specifying the `opaque` property in `UIGraphicsBeginImageContextWithOptions` in cd072e30.

What these changes will do is to automatically set the `opaque` flag to true when the image being resized is already opaque. In all the cases where transparent padding is used, we need to continue to set the `opaque` flag to `false`, otherwise that can result in undesired colored padding.

I also added some additional information to the public [docstring](https://github.com/Alamofire/AlamofireImage/blob/master/Source/UIImage%2BAlamofireImage.swift#L152-L165) of the `af_imageAspectScaledToFitSize` method to make it a bit easier to understand how the method behaves.

Hopefully these changes help make things easier for you.

Thanks again! 🍻
